onnxruntime/core/providers/webnn/builders/impl/gather_op_builder.cc

```diff
@@ -49,38 +47,18 @@ Status GatherOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   return Status::OK();
 }
 
-// Operator support related.
-
-bool GatherOpBuilder::IsOpSupportedImpl(const GraphViewer&,
-                                        const Node& node,
-                                        const WebnnDeviceType /* device_type */,
-                                        const logging::Logger& logger) const {
-  const auto& input_defs = node.InputDefs();
-  std::vector<int64_t> input_shape;
-  if (!GetShape(*input_defs[0], input_shape, logger))
-    return false;
-  const auto rank = input_shape.size();
-  if (rank < 1) {
-    LOGS(logger, VERBOSE) << "Gather only supports input shapes >= 1D, but input is "
-                          << rank << "d shape";
-    return false;
-  }
-
-  return true;
-}
-
```

Comment:
@Honry  I'm thinking these code might not be removed, I thought that the check logic in `IsInputRankSupportByOp` should check each op input so it should cover above check logic, but it did not check whether input_defs[index] exists.